pkg/controller/certificate-shim/lister.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	networkinglisters "k8s.io/client-go/listers/networking/v1beta1"
+	"k8s.io/client-go/tools/cache"
+	gatewaylisters "sigs.k8s.io/gateway-api/pkg/client/listers/apis/v1alpha1"
+)
+
+// objectLister is the most minimal generic lister.
```

Comment:
This isn't very nice. But I couldn't think of anything better

Reply:
After thinking a lot, I could not think of any other way> I thought about injecting `Get() metav1.Object` and `List() []metav1.Object` but in the end the code would be very similar with what you have written in lister.go.

I approve `lister.go`!